src/prefect/engine/flow_runner.py

```diff
@@ -154,7 +154,7 @@ def initialize_run(  # type: ignore
             for param, value in parameters.items():
                 context_params[param] = value
 
-        context.update(flow_name=self.flow.name)
+        context.update(flow=self.flow, flow_name=self.flow.name)
```

Comment:
This is the only piece of the PR that makes me hesitate -> Flow objects could be quite large, and adding this to context guarantees that all Dask communication has to ship the Flow along with every task run submission.  Without a strong argument _for_ its inclusion, I vote to not include it for now until we better understand the implications.

It is definitely possible that Flows are already serialized since we submit a FlowRunner method to Dask, but this feels like something we shouldn't do off-the-cuff without thinking through it.